### PR TITLE
We 457 : Implement delete and checkable rows for TrajectoryStation

### PR DESCRIPTION
**Pull request by eirinsvi**
# Request Template Witsml Explorer
Thank you for contributing to WITSML Explorer!
Before submitting this PR, please fill in the following template.

## Fixes
This pull request fixes #we-457

## Description
Implemented functionality for checking and delete multiple trajectory stations.

## Type of change
_List the types of change. Remove from the list any points which are not necessary_

* New feature (non-breaking change which adds functionality)

## Impacted Areas in Application
_List general components of the application that this PR will affect:_

* Api
* Frontend

## Checklist:
_Please tick all the boxes or remove the ones that aren't needed and explain why_

*Communication*
* [x] PR is related to an issue

*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass

### CODE REVIEW

File: Src/WitsmlExplorer.Api/Jobs/DeleteTrajectoryStationsJob.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using WitsmlExplorer.Api.Jobs.Common;
+using System.Collections.Generic;
+
+namespace WitsmlExplorer.Api.Jobs
+{
+    public record DeleteTrajectoryStationsJob
+    {
+        public TrajectoryReference Trajectory { get; init; }
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Jobs/DeleteTrajectoryStationsJob.cs:8]: Use "TrajectoryStationReferences" here

File: Src/WitsmlExplorer.Api/Models/JobType.cs [hunk 2]
```diff
@@ -15,6 +15,8 @@ public enum JobType
         DeleteLogObjects,
         DeleteMnemonics,
         DeleteTrajectory,
+        DeleteTrajectoryStations,
+
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Models/JobType.cs]: empty line

File: Src/WitsmlExplorer.Api/Workers/DeleteTrajectoryStationsWorker.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            query = TrajectoryQueries.GetWitsmlTrajectoryById(wellUid, wellboreUid, trajectoryUid);
+            var queryResult = await witsmlClient.GetFromStoreAsync(query, new OptionsIn(ReturnElements.IdOnly));
+
+            var trajectory = queryResult.Trajectories.First();
+            EntityDescription description = null;
+            if (trajectory != null)
+            {
+                description = new EntityDescription
+                {
+                    //TODO: this might be wrong names
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Workers/DeleteTrajectoryStationsWorker.cs]: The repo seems to be split between "WellName" and "NameWell". Might as well disregard it for the current PR.

File: Src/WitsmlExplorer.Api/Jobs/DeleteTrajectoryStationsJob.cs (hunk 1 above)
Comment (janburak) [reply Src/WitsmlExplorer.Api/Jobs/DeleteTrajectoryStationsJob.cs:8]: Remember to also update the job accordingly in frontend

File: Src/WitsmlExplorer.Api/Workers/DeleteTrajectoryStationsWorker.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            witsmlClient = witsmlClientProvider.GetClient();
+        }
+
+        public override async Task<(WorkerResult, RefreshAction)> Execute(DeleteTrajectoryStationsJob job)
+        {
+            var wellUid = job.Trajectory.WellUid;
+            var wellboreUid = job.Trajectory.WellboreUid;
+            var trajectoryUid = job.Trajectory.TrajectoryUid;
+            var trajectoryStation = new ReadOnlyCollection<string>(job.Uids.ToList());
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Workers/DeleteTrajectoryStationsWorker.cs]: Use plural form "trajectoryStations" when working with lists

File: Src/WitsmlExplorer.Api/Workers/DeleteTrajectoryStationsWorker.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            var result = await witsmlClient.DeleteFromStoreAsync(query);
+            if (result.IsSuccessful)
+            {
+                Log.Information("{JobType} - Job successful", GetType().Name);
+                var refreshAction = new RefreshTrajectory(witsmlClient.GetServerHostname(), wellUid, wellboreUid, trajectoryUid, RefreshType.Update);
+                var workerResult = new WorkerResult(witsmlClient.GetServerHostname(), true, $"Deleted trajectoryStation: {trajectoryStationString} for trajectory: {trajectoryUid}");
+                return (workerResult, refreshAction);
+            }
+
+            Log.Error("Failed to delete trajectoryStation for trajectory object. WellUid: {WellUid}, WellboreUid: {WellboreUid}, Uid: {TrajectoryUid}, TrajectoryStation: {TrajectoryStationString}",
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Workers/DeleteTrajectoryStationsWorker.cs]: Also plural

File: Src/WitsmlExplorer.Api/Jobs/DeleteTrajectoryStationsJob.cs (hunk 1 above)
Comment (eirinsvi) [reply Src/WitsmlExplorer.Api/Jobs/DeleteTrajectoryStationsJob.cs:8]: do you mean:
public TrajectoryStationReferences Trajectory { get; init; }
or
public TrajectoryStationReferences TrajectoryStation { get; init; }
?

**Comment by steinsiv** [comment]
Tested locally, delete functionality worked nicely 👍  
Reviewed code with nothing to add to earlier reviews

**Review by stokkoff — DISMISSED** [review]

**Review by janburak — APPROVED** [review]
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Pull request merged**